### PR TITLE
Make redirects permanent

### PR DESCRIPTION
**Pull request by am-on**
Change HTTP status code of redirects to 301.

### CODE REVIEW

**Review by dmurko — APPROVED** [review]

**Comment by dmurko** [comment]
302 gets indexed, 301 should not as per below sources.

https://www.quora.com/Why-is-Google-indexing-301-redirection-pages
https://moz.com/community/q/google-is-indexing-my-301-redirects-to-other-sites

**Comment by zupo** [comment]
302 is dangerous because browsers cache them. If you muck them up, there is no way to fix it other than telling all your visitors to clear their entire browser cache (which is impossible). 

There needs to be a way to specify which type of a redirect you want to do.

**Review by zupo — CHANGES_REQUESTED** [review]
301 are dangerous because browsers cache them. If you muck them up, there is no way to fix it other than telling all your visitors to clear their entire browser cache (which is impossible).

There needs to be a way to specify which type of a redirect you want to do. Then you always do 302 first, leave it in production for a week, then switch over to 301.

**Comment by dmurko** [comment]
If it helps: I don't see a use case for temporary (302) redirect, and we've never used it intentionally in the past.

File: pyramid_redirect/__init__.py [hunk 1]
```diff
@@ -81,7 +81,10 @@ def redirect_tween(request):
                     logger = logging.getLogger(__name__)
                     logger.info('Redirecting url: %s --> %s' \
                         % (request.url, url))
-                return HTTPFound(url)
+                if perament:
```
Comment (zupo) [inline pyramid_redirect/__init__.py]: typo `perament`

**Comment by zupo** [comment]
@am-on: please add docs that `permanent` redirects are dangerous because they get stored in visitors local browser caches that cannot be purged so if you do them wrong you are in pile of trouble. Recommend people use 302 for a few days and only then switch over to 301.

**Comment by zupo** [comment]
Mixed up 302 and 301. Fixed my comments.

**Comment by am-on** [comment]
@zupo It looks like browsers stopped caching 301 redirects. 
>Tested 301 redirect caching 02/2019 using Chrome 72, Firefox 65, UC Browser 7, Chromium 41 (for no real reason other than I had it installed!), latest version of Chrome on Android, latest version of Safari on iPhone, IE11 and Edge.
>
>None of the browsers cached the 301s!
>https://www.epiphanysearch.co.uk/thoughts/do-modern-browsers-still-cache-301-redirects

**Comment by zupo** [comment]
@am-on: do we still need this?

**Comment by am-on** [comment]
@zupo would be nice to have for minisites

**Comment by zupo** [comment]
I'll cut a release then.

**Comment by zupo** [comment]
https://pypi.org/project/pyramid_redirect/0.3/

**Pull request merged**